app/worker/utils.go

```diff
@@ -15,17 +15,57 @@
 package worker
 
 import (
-	"fmt"
+	"os"
+	"regexp"
 	"strings"
 
 	"github.com/mendersoftware/workflows/model"
+	"github.com/thedevsaddam/gojsonq"
 )
 
+const (
+	workflowEnvVariable   = "env."
+	workflowInputVariable = "workflow.input."
+)
+
+var reExpression = regexp.MustCompile(`\$\{([^\}]+)\}`)
```

Comment:
real nitpicking mode on: this can be fooled by putting into the data (workflow body)`'}'` character. This can in theory be a complex case of parsing.

Reply:
I'm not sure I get it, what's the problem if you put `'}'` into the workflow body?

Reply:
no wait. disregard. it is not as easy as I described. it should be fine.